katsdpsigproc/test/test_accel.py

```diff
@@ -331,53 +343,55 @@ def test_raw(self, context, queue):
 class TestPinnedAMD(TestDeviceArray):
     """Run DeviceArray tests forcing `_PinnedAMD` class for pinned memory."""
     @device_test
-    def setup(self, context, device):
-        context._force_pinned_amd = True
+    def setup(self, context: AbstractContext, queue: AbstractCommandQueue) -> None:
+        context._force_pinned_amd = True        # type: ignore
         super(TestPinnedAMD, self).setup()
 
     @device_test
-    def teardown(self, context, device):
-        context._force_pinned_amd = False
+    def teardown(self, context: AbstractContext, queue: AbstractCommandQueue) -> None:
+        context._force_pinned_amd = False       # type: ignore
 
 
 class TestSVMArrayHost(TestHostArray):
     """Tests SVMArray using the HostArray tests"""
+
     cls = SVMArray
 
-    def allocate(self, shape, dtype, padded_shape):
+    def allocate(self, shape: Tuple[int, ...], dtype: np.dtype,
+                 padded_shape: Tuple[int, ...]) -> SVMArray:
         return self.cls(self.context, shape, dtype, padded_shape)
 
     @device_test
     @cuda_test
-    def setup(self, context, queue):
+    def setup(self, context: AbstractContext, queue: AbstractCommandQueue) -> None:
         self.context = context
         super(TestSVMArrayHost, self).setup()
 
 
 class TestSVMArray(TestDeviceArray):
     """Tests SVMArray using the DeviceArray tests, plus some new ones"""
+
     cls = SVMArray
 
-    def _allocate_raw(self, context, n_bytes):
+    def _allocate_raw(self, context: AbstractContext, n_bytes: int) -> object:
         return context.allocate_svm_raw(n_bytes)
 
     @device_test
     @cuda_test
-    def setup(self, context, queue):
+    def setup(self, context: AbstractContext, queue: AbstractCommandQueue) -> None:
         super(TestSVMArray, self).setup()
 
     @device_test
     @cuda_test
-    def test_coherence(self, context, queue):
-        """Check that runtime correct transfers data between the host and
-        device views."""
+    def test_coherence(self, context: AbstractContext, queue: AbstractCommandQueue) -> None:
+        """Check that runtime correct transfers data between the host and device views."""
         source = """\
             <%include file="/port.mako"/>
             KERNEL void triple(unsigned int *data)
             {
                 data[get_global_id(0)] *= 3;
             }"""
-        program = accel.build(context, None, source=source)
+        program = accel.build(context, 'not a file', source=source)
```

Comment:
Why this change?

Reply:
mypy was complaining because the parameter is annotated as `str` and `None` isn't a `str`.